File: app/src/main/java/com/softaai/designpatterns/java/factory/ShapeFactory.java
package com.softaai.designpatterns.java.factory;

/**
 * Created by Amol Pawar on 27-03-2019.
 * softAai Apps
 */
public class ShapeFactory {

    public Shape getShape(String shapeType){
        if(shapeType == null){
            return null;
        }
        if(shapeType.equalsIgnoreCase("RECTANGLE")){
            return new Rectangle();
        }
        else if(shapeType.equalsIgnoreCase("SQUARE")){
            return new Square();
        }
        else if(shapeType.equalsIgnoreCase("CIRCLE")){
            return new Circle();
        }
        return null;
    }
}

File: app/src/main/java/com/softaai/designpatterns/kotlin/creational/factory/ShapeFactory.kt
package com.softaai.designpatterns.kotlin.creational.factory


/**
 * Created by Amol Pawar on 28-03-2019.
 * softAai Apps
 */
class ShapeFactory {
    fun getShape(shapeType: String) : Shape? {

        if(shapeType.equals("Rectangle", ignoreCase = true)){
            return Rectangle()
        }
        else if(shapeType.equals("Square", ignoreCase = true)){
            return Square()
        }
        else if(shapeType.equals("Circle", ignoreCase = true)){
            return Circle()
        }

        return null
    }
}